src/axom/slic/docs/sphinx/sections/getting_started.rst

```diff
@@ -0,0 +1,243 @@
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _sections/getting_started:
+
+Getting Started with Slic
+--------------------------
+
+This section illustrates some of the key concepts and capabilities of Slic by
+presenting a short walk-through of a C++ application. The complete
+:ref:`SlicApplicationCodeExample` is included in the
+:ref:`sections/appendix` section and is also available within the Axom source
+code, under ``src/axom/slic/examples/basic/logging.cpp``.
+
+This example illustrates the following concepts:
+
+* Initializing the Slic Logging Environment,
+* Prescribing the :ref:`logMessageFormat`,
+* Basic logging to the console using the :ref:`GenericOutputStream`, and,
+* Using some of the various :ref:`sections/slic_macros` to log messages.
+
+Step 1: Add Header Includes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+First, the Slic header must be included to make all the Slic
+functions and classes accessible to an application:
+
+.. literalinclude:: ../../../examples/basic/logging.cpp
+   :start-after: SPHINX_SLIC_INCLUDES_BEGIN
+   :end-before: SPHINX_SLIC_INCLUDES_END
+   :language: C++
+   :linenos:
+
+.. note::
+
+   All the classes and functions in Slic are encapsulated within the
+   ``axom::slic`` namespace.
+
+Step 2: Initialize Slic
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Prior to logging any messages, the Slic Logging Environment is initialized
+by the following:
+
+.. literalinclude:: ../../../examples/basic/logging.cpp
+   :start-after: SPHINX_SLIC_INIT_BEGIN
+   :end-before: SPHINX_SLIC_INIT_END
+   :language: C++
+   :linenos:
+
+This creates the root logger instance. However, in order to log messages,
+an application must first specify an output destination and optionally,
+prescribe the format of the log messages. These steps are demonstrated in
+the following sections.
+
+.. _slicExampleStep3:
+
+Step 3: Set the Message Format
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :ref:`logMessageFormat` is specified as a string consisting of keywords,
+encapsulated in ``< ... >``, that Slic knows how to interpret when assembling
+the log message.
+
+.. literalinclude:: ../../../examples/basic/logging.cpp
+   :start-after: SPHINX_SLIC_FORMAT_MSG_BEGIN
+   :end-before: SPHINX_SLIC_FORMAT_MSG_END
+   :language: C++
+   :linenos:
+
+For example, the ``format`` string in the code snippet above indicates that
+the resulting log messages will have the following format:
+
+* A line with the message time stamp
+* A line consisting of the :ref:`logMessageLevel`, enclosed in
+  brackets ``[ ]``, followed by the user-supplied message,
+* A third line with the name of the file where the message was emitted and
+* The corresponding line number location within the file, in the fourth line.
+
+See the :ref:`logMessageFormat` section for the complete list of keyword options
+available that may be used to customize the format of the messsages.
+
+.. note::
+
+   This step is optional. If the format is not specified, a default format
```

Comment:
I think you can omit the part starting with ", but, ....". Also, it would be helpful to briefly explain the default message format so users can see if it meets their needs -- perhaps in a short section by itself?